### PR TITLE
Add Hogwarp

### PR DESCRIPTION
**Pull request by ImKringle**
# Description

Added Hogwarp egg, a server that utilizes the Hogwarp mod for Multiplayer in Hogwarts Legacy
- Also handles #2198

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

Docs: https://docs.hogwarp.com/hogwarp/server/quick-start

Im new to handling Github so sorry for the messy commits - I can re-push req this if needed <3

### CODE REVIEW

**Comment by ImKringle** [comment]
To note, per the issue mentioned:
- Docs are semiavailable now
- Code will be open source on full release
- Requires 0 payment for use (as 4 player servers are free now)

File: game_eggs/hogwarp/egg-hogwarp.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string",
+            "field_type": "text"
+        },
+        {
+            "name": "Download URL",
+            "description": "URL to pull the files from\r\n- Using GDrive for no bandwidth limits",
+            "env_variable": "DOWNLOAD_URL",
+            "default_value": "http:\/\/dl.dropboxusercontent.com\/scl\/fi\/cd47ho7mhcvt91glm9q6q\/Hogwarp-Server-Files.zip?rlkey=juq8ru988l1yhuc16cwhx2jho&dl=1",
```
Comment (parkervcp) [inline game_eggs/hogwarp/egg-hogwarp.json:105]: I am going with a no for this alone. Is this how they distribute the files or is this an file from you?

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: game_eggs/hogwarp/egg-hogwarp.json (hunk 1 above)
Comment (ImKringle) [reply game_eggs/hogwarp/egg-hogwarp.json:105]: Its just how I was able to get the Download URL to be a direct link - Did you have a sugg for how to run the files?

Theyre distributed through their discord which makes it pulling straight from there a bit of a nightmare

**Comment by ImKringle** [comment]
Removed the DOWNLOAD_URL entry as needed + Noted the discord invite and the channel its in for any egg users!

**Comment by ImKringle** [comment]
Updated the egg slightly - Just not pushing here unless its alright! - Using the discord download link for the zip - provided by the dev team in that version; just waiting on WineTricks for plugin useability (Dotnet is so fun)

**Comment by parkervcp** [comment]
The link being from discord is better I guess. I don't see why they can't just host it on their site they built out for all of this.

**Comment by ImKringle** [comment]
Truthfully me neither - I'm hoping itll get hosted somewhere eventually but for now its literally just a zip on their discord 🤷🏼 even a git would be easier lol

**Review by QuintenQVD0 — APPROVED** [review]
LGTM!

**Pull request merged**